### PR TITLE
Enh flash sinter

### PR DESCRIPTION
**Pull request by tacaswell**

### CODE REVIEW

**Comment by CJ-Wright** [comment]
How are the ophyd objects here hinted? What are the expected independent variables of the scan?

File: startup/95-flash.py [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+
+        # take a measurement on the way in
+        yield from bps.trigger_and_read(all_dets)
+
+    return (yield from flash_step_field_inner())
+
+
+def flash_ramp(dets, start_I, stop_I, ramp_rate, md, *,
+               delay=1, mm_mode='Current', per_step=None):
+    start_V = 200
```
Comment (CJ-Wright) [inline startup/95-flash.py]:
```suggestion
    start_V = 200
    if per_step is None:
        per_step = bps.trigger_and_read
```

File: startup/95-flash.py [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+    @run_decorator(md=md)
+    def flash_step_field_inner():
+        # set everything to zero at the top
+        yield from bps.mv(flash_power.current_sp, 0,
+                          flash_power.voltage_sp, 0,
+                          flash_power.ramp_rate, ramp_rate)
+        # put in "Duty Cycle" mode so current changes immediately
+        yield from bps.mv(flash_power.mode, 'Duty-Cycle')
+        # take one shot on the way in
+        yield from bps.trigger_and_read(all_dets)
```
Comment (CJ-Wright) [inline startup/95-flash.py]:
```suggestion
        yield from per_step(all_dets)
```

File: startup/95-flash.py [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+        yield from bps.mv(flash_power.mode, 'Duty-Cycle')
+        # set the target to let it go
+        yield from bps.mv(flash_power.current_sp, start_I)
+
+        yield from _inner_loop(dets, exposure_count,
+                               time.monotonic() + expected_time,
+                               delay, per_step)
+
+        # take one shot on the way out
+        yield from bps.trigger_and_read(all_dets)
```
Comment (CJ-Wright) [inline startup/95-flash.py]:
```suggestion
        yield from per_step(all_dets)
```

**Comment by tacaswell** [comment]
I thought the plan was to handle dark collection via a combination of decorators and pre-processors?

**Comment by CJ-Wright** [comment]
It might be better to be explicit here by putting it into the per_step. To some extent this depends on what @sghose had in mind here for the dark collection, should we collect darks during the scan, at the beginning of the scan, intermittently, etc

**Comment by awalter-bnl** [comment]
> I thought the plan was to handle dark collection via a combination of decorators and pre-processors?

I agree with @tacaswell here, we have an approach to taking darks that is flexible (using decorators and re-processors) and should be using that rather than introducing another method. If the exact approach for deciding _when_ to take darks is not in the existing method we should add it there.

**Comment by tacaswell** [comment]
Running this through `xrun` picks up at least the taking of dark frame logic.  Did not chace it through to make sure it works in the analysis pipeline.

Mostly a success!  I think the users will be able to take productive data this weekend.

Did not test the x-motor related per-step functions.

**Pull request merged**